src/rebalancers/uniswap-v3/UniswapV3Rebalancer.sol

```diff
@@ -0,0 +1,487 @@
+
+        uint256 newTokenId;
+        (newTokenId,, amount0, amount1) = UniswapV3Logic.POSITION_MANAGER.mint(
+            MintParams({
+                token0: position.token0,
+                token1: position.token1,
+                fee: position.fee,
+                tickLower: position.newLowerTick,
+                tickUpper: position.newUpperTick,
+                amount0Desired: balance0,
+                amount1Desired: balance1,
+                amount0Min: 0,
+                amount1Min: 0,
+                // TODO: send direct to actionHandler ?
+                recipient: address(this),
+                deadline: block.timestamp
+            })
+        );
+
+        // Update the tokenId for newly minted position
+        assetData.assetIds[0] = newTokenId;
+
+        // Any surplus is transferred to the Account.
+        balance0 -= amount0;
+        balance1 -= amount1;
+        if (balance0 > 0) ERC20(position.token0).safeTransfer(account_, balance0);
+        if (balance1 > 0) ERC20(position.token1).safeTransfer(account_, balance1);
+
+        // Approve ActionHandler to deposit Liquidity Position back into the Account.
+        UniswapV3Logic.POSITION_MANAGER.approve(msg.sender, newTokenId);
+    }
+
+    /* ///////////////////////////////////////////////////////////////
+                        INITIATORS LOGIC
+    /////////////////////////////////////////////////////////////// */
+    /**
+     * @notice Sets an initiator for an Account. An initiator will be permisionned to rebalance any UniswapV3
+     * Liquidity Position held in the specified Arcadia Account.
+     * @param initiator The address of the initiator.
+     * @param account_ The address of the Arcadia Account to set an initiator for.
+     */
+    function setInitiatorForAccount(address initiator, address account_) external {
+        ownerToAccountToInitiator[msg.sender][account_] = initiator;
+    }
+
+    /**
+     * @notice Sets the information requested for an initiator.
+     * @param fee . TODO
+     * @param tolerance The maximum deviation of the actual pool price,
+     * relative to the price calculated with trusted external prices of both assets, with 18 decimals precision.
+     * @dev The tolerance for the pool price will be converted to an upper and lower max sqrtPrice deviation,
+     * using the square root of the basis (one with 18 decimals precision) +- tolerance (18 decimals precision).
+     * The tolerance boundaries are symmetric around the price, but taking the square root will result in a different
+     * allowed deviation of the sqrtPriceX96 for the lower and upper boundaries.
+     */
+    function setInitiatorInfo(uint256 tolerance, uint256 fee) external {
+        // Cache struct
+        InitiatorInfo memory initiatorInfo_ = initiatorInfo[msg.sender];
+
+        if (initiatorInfo_.fee > 0 && fee > initiatorInfo_.fee) revert DecreaseFeeOnly();
```

Comment:
I'll have to find a way to initialize fee and remove "initiatorInfo_.fee > 0", as now an initiator can set fee to 0 and then to max again. Double check flow for tolerance below as well.